### ISSUE TITLE
Drupal 8.6/8.7 Documentation Update

### ISSUE THREAD

**Issue by ianlintner** [phase2/particle#543]
I ran through the Drupal 8 Quick start guide. The biggest problem I ran into was the Components module error'd out when the twig templates did not exist. I had to update the index.js  -- note I haven't been through then entire guide at this point or misread it.

Component library was causing the error:

```Uncaught PHP Exception Twig\Error\LoaderError: "The "themes/particle/apps/drupal/../../dist/app-drupal/assets/atomic/_patterns/00-protons" directory does not exist ("/var/www/html/web/themes/particle/apps/drupal/../../dist/app-drupal/assets/atomic/_patterns/00-protons")```

My Updates to the index.js

Make sure all the atomic levels were included in the atomic context.
```
const atomicContext = require.context(
  // From patterns folder
  './_patterns',
  // Deep dive all directories below
  true,
  // Get the first folders after atoms|molecules|organisms
  /^\.\/(00-protons|01-atoms|02-molecules|03-organisms)\/[\w-]+$/
);
```

Uncomment the following sections (Which I assumed they are optional for standard non-drupal build): 
```
importAll(
   require.context(
     'templates',
     true,
     /^\.\/[\w-]+$/ // See note on static regex
   )
 );
// Pages. Skipping for design system. Include per-app.
 importAll(
   require.context(
     'pages',
     true,
     /^\.\/[\w-]+$/ // See note on static regex
   )
 );
```

If there is a better way or other changes needed after running the Drupal Build please let me know. -- BTW This is a great project.

**Comment by ianlintner** [comment]
Either way let me know -- If I have time I can try to submit a PR for updated docs. I am working on getting a build process up and running for a team -- so I need to do the documentation work one way or another.

**Comment by ocjojo** [comment]
Had the same problem with a subfolder of `00-protons`
I fixed it in the namespace-writer. See attached patch. Let me know if I should make this a PR.
[fix_namespaces_for_drupal.txt](https://github.com/phase2/particle/files/3147749/fix_namespaces_for_drupal.txt)

**Comment by jryanconklin** [comment]
Indeed! We'll need to fix the namespace maker to make sure it only updates the info file for directories. Thank you!

**Comment by p51i** [comment]
@ocjojo Thanks for this fix. We are using it for the same issue. One thing - `fs.mkdirSync` was failing for us on first build of the dist directory. This tweak seems to be working (so far):


```
// Each app provides its own transform function :)
const relativePath = config.namespaces.transform(atomicPath)
const absolutePath = path.resolve(config.APP_PATH, relativePath)
if (!fs.existsSync(absolutePath)) {
  fs.mkdir(absolutePath, {recursive: true}, err => {});
}
```
Thanks again. Hope this helps!

**Comment by jryanconklin** [comment]
@p51i Howdy! If you have a fix for this issue, would you mind PR'ing it? That'd help quite a bit.

**Comment by p51i** [comment]
@jryanconklin I thought I had something, but I think you are correct in that the answer lies in keeping unused paths out of info. `recursive` only works on node >= 10.2. That may not be an issue for all, but it seems limiting (such as: my build container). I'm still trying a few things. If I make anything work better I promise I'll submit a PR!

**Comment by p51i** [comment]
This seems related? At least partially. https://github.com/phase2/particle/pull/525/files

**Comment by p51i** [comment]
Duplicate of #503 and is fixed with #525 

`particle.info.yml` only contains paths with existing files and it works like a champ.

**Comment by jryanconklin** [comment]
@p51i Thanks Mark! I think there's an issue with #525, but I set a meeting up today to see what feedback it needs or if we need a different fix.

**Comment by jryanconklin** [comment]
Just wanted to say thanks again for this issue and your involvement everyone. We have a short term fix in mind (skip those folders on namespace write). A longer term fix is going to be only writing those out based on real Webpack emitted output (preferred). We'll go with the latter assuming we can get it to work on the shorter time frame spectrum. Stay tuned!

**Comment by Nebel54** [comment]
@jryanconklin Thanks to the whole particle theme for this great project!

As the author of #525 I would love to get an idea of what kind of issues my PR raises. Could you go a bit more into details? Thank you!

**Comment by jryanconklin** [comment]
@Nebel54 Thank you so much for bringing this up and the PR! We've decided to fix this in a less heavy manner in #552. Thought here being that this is what's really required for current version of Component Library, e.g., skip unresolved directory paths. This is a quick step to get folks unblocked since Component Library 8.x-1.1 is officially out. 

Issue #554 has been opened for the longer term goal of moving namespace generation to Webpack bundled output.

CC: @illepic
